app/routes/synonyms/synonyms.server.ts

```diff
@@ -166,6 +166,55 @@ export async function moderatorCreateSynonyms(synonymousEventIds: string[]) {
   return uuid
 }
 
+export async function deleteIfGroupIsEmpty(eventId: string) {
+  const db = await tables()
+  const circulars = await getSynonymMembers(eventId)
+
+  if (circulars.length === 0) {
+    await db.synonyms.delete({ eventId })
+  }
+}
+
+export async function manageSynonymVersionUpdates(
+  newEventId: string,
+  oldEventId?: string,
+  createdOn?: number
+) {
+  if (newEventId === oldEventId) return
+
+  const oldestDate = await getOldestDate(newEventId)
+  const dateParam = oldestDate ? oldestDate : createdOn
+  if (!dateParam) throw Error
+
+  const promises = []
+  if (newEventId) {
+    promises.push(tryInitSynonym(newEventId, dateParam))
+  }
+  if (oldEventId && oldEventId != newEventId) {
+    promises.push(deleteIfGroupIsEmpty(oldEventId))
+    promises.push(updateInitialDate(oldEventId))
```

Comment:
You're doing more database work than you need to here: if the group is empty, then you need to delete it; otherwise, you need to update the date.

Also, where do you update the date on the group containing the _new_ event ID?